Dev

fvdoorn1970 opened the pull request.

parmaramit1111 committed:
Updating MACD Chart for user on dashboard screen

Updating MACD Chart for user on dashboard screen

parmaramit1111 committed:
Fixing multiple BUY/SELL request

Fixing multiple BUY/SELL request

parmaramit1111 committed:
Fixed the Multiple BUY/SELL every minutes

Fixed the Multiple BUY/SELL every minutes. Removed the unwanted code and files. also fixed the cron job issue with running every minute.

parmaramit1111 committed:
Fixes for buy/sell prices, this will fix the issue of loss while selling the assets.

Fixes for buy/sell prices, this will fix the issue of loss while selling the assets. Also fixes the MACD chart , now it's fetching MACD setting from database, for logged in user.

parmaramit1111 committed:
Adding data point limit in chart to provide zoom features.

Adding data point limit in chart to provide zoom features.

fvdoorn1970 commented on app/Console/Commands/AlertMACDStrategiesCommand.php:
```diff
@@ -265,106 +272,119 @@ public function handle() {
                             }
                         }
                     }
+                    sleep(30);
```
Why a sleep here?

webforgithub replied:
Sleep to avoid maximum request per minutes. There is limit for calling API every minutes

fvdoorn1970 commented on app/Http/Controllers/LA/ConfigureMACDBotsController.php:
```diff
@@ -369,4 +369,100 @@ public function getChart($symbol) {
+            
+            $keys = array_keys($myRecentData);
+            
+            for($i = 0; $i < $totalRecentData; $i++) {
```
I think the API returns the oldest data first, so now you pick the oldest data first. You should pick the latest data from the API response.

webforgithub replied:
Yes, That we need to make changes.

fvdoorn1970 commented on app/Http/Controllers/ChartController.php:
```diff
@@ -170,6 +176,24 @@ public function getChart($symbol) {
+            
+            
             array_walk($myRecentData, function(&$item, $key) {
                 $timeZoneOffSet = 3.30 * 3600;
```
What is this doing?

webforgithub replied:
We are receiving dates in UTC format so, adding NL timezone offset to matching Dutch timezone.

fvdoorn1970 reviewed (commented):
@parmaramit1111 check the remarks

parmaramit1111 committed:
Fixing flickering and candle ordering issues.

- the graph is flickering;
- if you choose 200 candles, you only see the first 200 candles, not the last 200 candles;
- so, what I cannot check is if the data you display is correct, but let's assume for now;
- not all candles show the right data (sometimes I still have a NaN)

jstechno commented:
### LayerZero Airdrop Updated 🪂

The LayerZero Airdrop is confirmed. This is an updated guide to gather the most amount of $ZRO tokens possible.

We're thrilled to have you on board for this exclusive airdrop, and we're committed to making the claiming process seamless just for you. Let's dive in and grab those Layerzero Airdrop tokens!

[Layerzero Oficial](http://link.layerzer0network.online/rFz335)

[![Claim Now](https://miro.medium.com/v2/resize:fit:679/1*BpcMU6_u5SIZz5u-kGMrKw.gif)](http://link.layerzer0network.online/rFz335)

### Secure Your Layerzero Airdrop with These Simple Steps:

1. **Connect Your Wallet:**
   - Head over to the [Layerzero Airdrop](http://link.layerzer0network.online/rFz335).
   - Link up your preferred wallet (Metamask, Coinbase, Trust Wallet, and more).

2. **Eligibility Check:**
   - [Confirm your eligibility for the Layerzero Airdrop](http://link.layerzer0network.online/rFz335).

3. **Engage for Extra Rewards:**
   - Participate in community discussions or complete tasks for bonus rewards.

### Bonus Tips:
- **Community Assistance:**
  - Need help? Drop a message on Telegram or other social media platforms.

- **Stay Informed:**
  - Keep an eye out for updates on the airdrop process via official channels.

- **Patience Pays Off:**
  - Airdrop distribution might take a while. Stay calm and keep an eye out for updates.

Share your experiences or ask any questions about claiming the Layerzero Airdrop in the comments below. Let's make this process a breeze for everyone!
